Functions/Functions/User/RemoveUserCommentsAccount.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using OidcApiAuthorization.Abstractions;
+using OidcApiAuthorization.Models;
+using System.Linq;
+
+namespace SSW.Rules.Functions
+{
+    public class RemoveUserCommentsAccount
+    {
+        private readonly RulesDbContext _dbContext;
+        private readonly IApiAuthorization _apiAuthorization;
+
+        public RemoveUserCommentsAccount(RulesDbContext dbContext, IApiAuthorization apiAuthorization)
+        {
+            _dbContext = dbContext;
+            _apiAuthorization = apiAuthorization;
+        }
+
+        [FunctionName("RemoveUserCommentsAccount")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+
+            ApiAuthorizationResult authorizationResult = await _apiAuthorization.AuthorizeAsync(req.Headers);
+
+            if (authorizationResult.Failed)
+            {
+                log.LogWarning(authorizationResult.FailureReason);
+                return new UnauthorizedResult();
+            }
+
+            log.LogWarning($"HTTP trigger function {nameof(RemoveUserCommentsAccount)} request is authorized.");
+
+            User data;
+
+            string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+
+            log.LogInformation(requestBody);
+
+            data = JsonConvert.DeserializeObject<User>(requestBody);
+
+            log.LogInformation(data.ToString());
+
+            if (data == null || string.IsNullOrEmpty(data?.UserId))
```

Comment:
data.ToString() will throw an exception before you hit this line if data is null.